thesis-de/content-de.tex

```diff
@@ -2715,8 +2722,123 @@ \section{Realisierung Grafana}
+und stellt verschiedene virtuelle Maschinen und Container bereit, in denen
+Tests ausgeführt werden können. GitHub kann bei neuen Pull Requests eine
+Benachrichtigung an Travis schicken. Daraufhin startet Travis eine neue VM und
+klont das entsprechende GitHub Repository und den \texttt{Feature Branch}.
+Danach wird eine definierte Testmatrix ausgeführt. Travis schickt eine
+Benachrichtigung an GitHub, nachdem die Tests beendet wurden. Somit ist in der
+Weboberfläche ersichtlich, ob die Tests erfolgreich waren oder nicht.
+
+Dieses Verfahren ermöglicht es, eine reproduzierbare Testumgebung zu nutzen.
+Diese wird immer auotmatisch gestartet. Es ist nicht mehr notwendig, jeden Test
+lokal auszuführen. Travis stellt VMs bereit, in denen nur das nötigste
+installiert ist. Dadurch ist sichergestellt, dass keine \glslink{Side
+Effect}{Side Effects} auftreten. Bei lokalen Tests tritt dies häufig auf. Der
+Entwickler hat diverse Bibliotheken installiert, welche von der entwickelten
+Software benötigt werden.  Innerhalb der entwickelten Software wird eine
+Abhängigkeit nicht definiert, es funktioniert aber trotzdem, da die Bibliothek
+zufällig vorhanden ist.  Innerhalb von Travis fällt dies aber auf. Außerdem ist
+es jedem Teammitglied möglich, die Travis Logs von überall aus zu begutachten.
+Alternativ müsste jeder Entwickler die Tests selbst lokal ausführen.
+\tm%
+
+\section{Linter und Syntax Validator}
+
+\section{Unit Tests}
+Für die Tests und Qualitätsicherung wurden unter anderem Unit Tests verwendet.
+Unit Tests sind Software Tests und werden immer dann verwendet, wenn jeweils
+die Kernfunktion von einem Quellcode geprüft werden soll. Das Framework hierzu
+ist von rspec entwickelt worden. Rspec besitzt bereits schon eine Reihe an
+Test, welche für die im Projekt verwendeten Profile entweder erweitert oder neu
+erstellt worden sind. Man kann hier mit einem Test, zum Beispiel prüfen, ob das
+Profil auf unterschiedlichen Betriebssystemen ausgeführt werden kann oder ob
+bestimmte Ressourcen im kompilierten Katalog enthalten sind. Zu Beginn müssen
+diese vorab mit rspec in dem katalog kompiliert und definiert werden, damit
+diese verwendet werden können. Mit folgendem Befehl, können die Tests
+durchgeführt werden:
+\begin{lstlisting}[language=bash]
+bundle exec rake spec
+\end{lstlisting}
+
+Im Beispiel im Grafana Profil, sehen die Tests wie folgt aus:
 
-\section{Tests und Qualitätssicherung}
+\begin{listing}
+  \inputminted{ruby}{../listings/rspec-grafana.txt}
+  \caption{Tests für das Grafana Profil}
+\end{listing}
+
+Zuerst werden verschiedene Betriebssysteme simuliert, um das Grafana Profil auf
+diesen zu testen. Jedes Betriebssystem besitzt betriebssystemspezifische
+\glslink{Fact}{Facts}. Diese werden von rspec simuliert und mit hilfe einer
+liste der unterstützten Betriebssystemen auf dem entsprechenend Profil
+getestet. Diese Liste an Betriebssystemen, wird aus der "metadata.json"
```

Comment:
metadata.jsn aus meinem PR sollte wohl in als listing in den anhang und hier verlinkt werden.